htdocs/core/class/hookmanager.class.php

```diff
@@ -219,7 +219,7 @@ function executeHooks($method, $parameters=array(), &$object='', &$action='')
 
                     $this->resNbOfHooks++;
 
-                    $modulealreadyexecuted[$module]=$module; // Use the $currentcontext in method to avoid running twice
+                    $modulealreadyexecuted[$actionclassinstance]=$actionclassinstance; // Use the $currentcontext in method to avoid running twice
```

Comment:
Using a full object as a key and value when we just need a flag/semaphore is a memory waste. Does using  get_class($actionclassinstance) instead is ok for you ?

Reply:
@ias-ceo @eldy 
this break this check : 

```
// test to avoid running twice a hook, when a module implements several active contexts
if (in_array($module,$modulealreadyexecuted)) continue;
```

and this already check here no ?
```
// jump to next module/class if method does not exist
if (! method_exists($actionclassinstance,$method)) continue;
```